contracts/FIFOMarket.sol

```diff
@@ -182,29 +202,42 @@ contract FIFOMarket is
       "Market: This contract can only receive BridgedPolygonNORI"
     );
 
-    uint256[] memory ids = new uint256[](_queueLength());
-    uint256[] memory amounts = new uint256[](_queueLength());
-    address[] memory suppliers = new address[](_queueLength());
+    uint256[] memory ids = new uint256[](totalNumberActiveRemovals);
+    uint256[] memory amounts = new uint256[](totalNumberActiveRemovals);
+    address[] memory suppliers = new address[](totalNumberActiveRemovals);
```

Comment:
the `totalNumberActiveRemovals` keeps track of the total number of removal ids that are for sale in the marketplace, and is basically the new "queue length" except for the fact that these ids are spread across multiple collections and so we can't get a simple `.length` on anything.  We still need this number in order to know the max number of iterations our main loop could take before concluding that there's not enough supply to fill the order